### PR TITLE
Backend shopping cart

### LINKED ISSUE #224: Implement shopping card functionality to the backend
Implement shopping card functionality to the backend

### PR DESCRIPTION
**Pull request by lighterm**
# Description

Implementation of edit and delete product endpoints.

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

Unit tests are implemented to test the new endpoints and the code passes all the existing and new tests.

**Test Configuration**:
* OS: MacOS
* Python: 3.8.5

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/shopping_cart/views.py [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+        product = product[0] 
+
+    try:
+        items = ShoppingCarts.objects.filter(Q(product=product_id, customer=customer))
+        if len(items) == 0:
+            item = ShoppingCarts.objects.create(product=product, customer=customer, quantity=quantity)
+            item.save()
+        else:
+            items[0].quantity = quantity
+            print("selam ben buraya geldim")
```
Comment (onurkilic2301) [inline backend/shopping_cart/views.py]: Print message can be removed

Comment (lighterm) [reply backend/shopping_cart/views.py]: Sorry about that I am removing it, it was for debugging. Fixed now.

**Review by onurkilic2301 — COMMENTED** [review]
Both add and delete endpoints raises value error if product_id is given as a non-numeric value i.e. product_id="asd". This can also be handled. Other than this the code works fine.

**Comment by lighterm** [comment]
- Implementation of helper functions (all categories, all vendors, all brands) added.
- "Both add and delete endpoints raises value error if product_id is given as a non-numeric value i.e. product_id="asd". This can also be handled." is handled.
- Tests are implemented.

**Review by onurkilic2301 — APPROVED** [review]

**Pull request merged**